iceoryx_posh/test/moduletests/test_popo_used_chunk_list.cpp

```diff
@@ -0,0 +1,305 @@
+
+        iox::posix::Allocator memoryAllocator{m_memory.get(), MEMORY_SIZE};
+        memoryManager.configureMemoryManager(mempoolconf, &memoryAllocator, &memoryAllocator);
+    };
+
+    void TearDown() override{};
+
+    SharedChunk getChunkFromMemoryManager()
+    {
+        constexpr uint32_t PAYLOAD_SIZE{32U};
+        auto chunkSettingsResult =
+            iox::mepoo::ChunkSettings::create(PAYLOAD_SIZE, iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT);
+        EXPECT_FALSE(chunkSettingsResult.has_error());
+        if (chunkSettingsResult.has_error())
+        {
+            return nullptr;
+        }
+        auto& chunkSettings = chunkSettingsResult.value();
+        return memoryManager.getChunk(chunkSettings);
+    }
+
+    void createMultipleChunks(uint32_t numberOfChunks, std::function<void(SharedChunk&&)> testHook)
+    {
+        ASSERT_TRUE(testHook);
+        for (uint32_t i = 0; i < numberOfChunks; ++i)
+        {
+            testHook(getChunkFromMemoryManager());
+        }
+    }
+
+    void checkIfEmpty()
+    {
+        SCOPED_TRACE(std::string("Empty check"));
+        for (uint32_t i = 0; i < USED_CHUNK_LIST_CAPACITY; ++i)
+        {
+            EXPECT_TRUE(sut.insert(getChunkFromMemoryManager()));
+        }
+    }
+
+    MemoryManager memoryManager;
+
+    static constexpr uint32_t USED_CHUNK_LIST_CAPACITY{10U};
+    UsedChunkList<USED_CHUNK_LIST_CAPACITY> sut;
+
+  private:
+    static constexpr size_t MEGABYTE = 1U << 20U;
+    static constexpr size_t MEMORY_SIZE = 4U * MEGABYTE;
+    std::unique_ptr<char[]> m_memory{new char[MEMORY_SIZE]};
+};
+
+TEST_F(UsedChunkList_test, OneChunkCanBeAdded)
+{
+    EXPECT_TRUE(sut.insert(getChunkFromMemoryManager()));
+}
+
+TEST_F(UsedChunkList_test, AddSameChunkTwiceWorks)
+{
+    auto chunk = getChunkFromMemoryManager();
+    sut.insert(chunk);
+
+    EXPECT_TRUE(sut.insert(chunk));
+}
+
+TEST_F(UsedChunkList_test, MultipleChunksCanBeAdded)
+{
+    EXPECT_TRUE(sut.insert(getChunkFromMemoryManager()));
+    EXPECT_TRUE(sut.insert(getChunkFromMemoryManager()));
+    EXPECT_TRUE(sut.insert(getChunkFromMemoryManager()));
+}
+
+TEST_F(UsedChunkList_test, AddChunksUpToCapacityWorks)
+{
+    createMultipleChunks(USED_CHUNK_LIST_CAPACITY, [this](SharedChunk&& chunk) { EXPECT_TRUE(sut.insert(chunk)); });
+}
+
+TEST_F(UsedChunkList_test, AddChunksUntilOverflowIsHandledGracefully)
+{
+    createMultipleChunks(USED_CHUNK_LIST_CAPACITY, [this](SharedChunk&& chunk) { EXPECT_TRUE(sut.insert(chunk)); });
+
+    EXPECT_FALSE(sut.insert(getChunkFromMemoryManager()));
+}
+
+TEST_F(UsedChunkList_test, OneChunkCanBeRemoved)
+{
+    auto chunk = getChunkFromMemoryManager();
+    auto chunkHeader = chunk.getChunkHeader();
+    sut.insert(chunk);
+
+    SharedChunk removedChunk;
+    EXPECT_TRUE(sut.remove(chunkHeader, removedChunk));
+    EXPECT_TRUE(removedChunk);
+
+    checkIfEmpty();
+}
+
+TEST_F(UsedChunkList_test, RemoveSameChunkAddedTwiceWorks)
```

Comment:
It should as it is implemented but we do not require this for our use case. Good to test anyway.